tests/components/building/datasets_test.py

```diff
@@ -113,6 +112,8 @@ def trajectory_dataset() -> TrajectoryDataset:
     return trajectory_dataset
 
 
+# TODO (sasha): not quite sure what to do for the transition dataset stuff because it
+# is now a numpy iterator and we cannot access the fields we want to test
 def test_init_transition_dataset(transition_dataset: TransitionDataset) -> None:
```

Comment:
I think we should keep a reference to the dataset so we can test here and just use the iterator when looping.

Reply:
Have tried this but it still doesn't seem to have any of the metadata. This is what the object looks like:

```
<ParallelInterleaveDataset element_spec=ReplaySample(info=SampleInfo(key=TensorSpec(shape=(), dtype=tf.uint64, name=None), probability=TensorSpec(shape=(), dtype=tf.float64, name=None), table_size=TensorSpec(shape=(), dtype=tf.int64, 
name=None), priority=TensorSpec(shape=(), dtype=tf.float64, name=None)), data=Transition(observations={'agent_1': OLT(observation=TensorSpec(shape=(10,), dtype=tf.float32, name=None), legal_actions=TensorSpec(shape=(), dtype=tf.int32,
 name=None), terminal=TensorSpec(shape=(), dtype=tf.int32, name=None)), 'agent_0': OLT(observation=TensorSpec(shape=(10,), dtype=tf.float32, name=None), legal_actions=TensorSpec(shape=(), dtype=tf.int32, name=None), terminal=TensorSpe
c(shape=(), dtype=tf.int32, name=None))}, actions={'agent_0': TensorSpec(shape=(), dtype=tf.int32, name=None), 'agent_1': TensorSpec(shape=(), dtype=tf.int32, name=None)}, rewards={'agent_0': TensorSpec(shape=(), dtype=tf.float32, nam
e=None), 'agent_1': TensorSpec(shape=(), dtype=tf.float32, name=None)}, discounts={'agent_1': TensorSpec(shape=(), dtype=tf.float32, name=None), 'agent_0': TensorSpec(shape=(), dtype=tf.float32, name=None)}, next_observations={'agent_
1': OLT(observation=TensorSpec(shape=(10,), dtype=tf.float32, name=None), legal_actions=TensorSpec(shape=(), dtype=tf.int32, name=None), terminal=TensorSpec(shape=(), dtype=tf.int32, name=None)), 'agent_0': OLT(observation=TensorSpec(
shape=(10,), dtype=tf.float32, name=None), legal_actions=TensorSpec(shape=(), dtype=tf.int32, name=None), terminal=TensorSpec(shape=(), dtype=tf.int32, name=None))}, extras={'extras': TensorSpec(shape=(), dtype=tf.float32, name=None)}
))>

```

Reply:
Discussed offline, going to just make sure it doesn't throw an error :smile: